wsireg/reg_images/np_reg_image.py

```diff
@@ -117,7 +136,7 @@ def read_single_channel(self, channel_idx: int):
             )
             channel_idx = 0
         if self.n_ch > 1:
-            if self.is_rgb:
+            if self.is_rgb and self.is_rgb_interleaved is True:
```

Comment:
Unless `self.is_rgb_interleaved` can be something other than `True/False` you can remove this explicit check.

```suggestion
            if self.is_rgb and self.is_rgb_interleaved:
```